x/slashing/genesis.go

```diff
@@ -12,7 +12,11 @@ import (
 func InitGenesis(ctx sdk.Context, keeper keeper.Keeper, stakingKeeper types.StakingKeeper, data *types.GenesisState) {
 	stakingKeeper.IterateValidators(ctx,
 		func(index int64, validator stakingtypes.ValidatorI) bool {
-			keeper.AddPubkey(ctx, validator.GetConsPubKey())
+			consPk, err := validator.TmConsPubKey()
```

Comment:
this doesnt need to be a tendermint key. I believe the only places that need to be tendermint keys is where it is going to Tendermint. Within the sdk you can use sdk's keys

Reply:
Why? this is a consensus key, and it's coming from Tendermint. I'm not very familiar with the tendermint side. I assume that when we get a tendermint key then we should use it consistently as a tendermint key except when we serialize it in SDK objects.

@marbar3778 could you confirm?

Reply:
Tendermint doesn't care how its handled. The reason you would want to handle it with your own key is to disentangle Tendermint from the inner workings of the sdk. You can use tendermint ed25519 keys here but then what is the point of having your own?

Reply:
To be able to serialize it in SDK structures / messages.